src/clients/QBittorrent.ts

```diff
@@ -215,76 +221,140 @@ export default class QBittorrent implements TorrentClient {
 	}
 
 	async isInfoHashInClient(infoHash: string): Promise<boolean> {
-		const responseText = await this.request(
-			"/torrents/properties",
-			`hash=${infoHash}`,
+		const torrent = await this.getTorrentInfo(infoHash);
+		return torrent.length > 0;
+	}
+
+	async addTorrent(formData: FormData): Promise<void> {
+		await this.request("/torrents/add", formData);
+	}
+
+	async recheckTorrent(infoHash: string): Promise<void> {
+		const torrent = await this.getTorrentInfo(infoHash);
+		if (torrent.length === 0) {
+			throw new Error("Torrent not found in client");
+		}
```

Comment:
given that this check requires an additional api call, maybe it's better to beg for forgiveness than ask for permission here by checking the result of the `recheck` api call?

Reply:
Do you mean to first check with the provided hash? Yeah that would work in most scenarios, only an issue for v2 which should be rare.

Reply:
I don't know of any PT's that are planning to implement v2 in any sort of haste, mostly no plans at all from what I've seen, however it is somewhat prevalent on certain public.

Given our limited support specifically for public, I'm not sure how much this is beneficial, however given that the check for v1 initially should catch most of these I think it isn't a bad idea to include compatibility.

Reply:
The check for all torrents is perfectly fine. It takes about 1 second with 6k torrents using `.fastresume`. With the refactoring I did, this is only called 3 times per inject:
1. Check if new torrent already exists
2. Confirm new torrent is added
3. If searchee has infoHash